unstructured_inference/inference/layoutelement.py

```diff
@@ -322,12 +328,53 @@ def get_elements_from_ocr_regions(ocr_regions: List[TextRegion]) -> List[LayoutE
             r.x2,
             r.y2,
             text=r.text,
+            source=None,
             type="UncategorizedText",
         )
         for r in merged_regions
     ]
 
 
+def separate(region_a: Union[LayoutElement, Rectangle], region_b: Union[LayoutElement, Rectangle]):
+    """Reduce leftmost rectangle to don't overlap with the other"""
+
+    def reduce(keep: Rectangle, reduce: Rectangle):
+        # Asume intersection
+
+        # Other is down
+        if reduce.y2 > keep.y2 and reduce.x1 < keep.x2:
+            # other is down-right
+            if reduce.x2 > keep.x2 and reduce.y2 > keep.y2:
+                reduce.x1 = keep.x2 * 1.01
+                reduce.y1 = keep.y2 * 1.01
+                return
+            # other is down-left
+            if reduce.x1 < keep.x1 and reduce.y1 < keep.y2:
+                reduce.x1 = keep.y2
```

Comment:
I think you meant to shrink down the `reduce` box in y direction, right?
```suggestion
                reduce.y1 = keep.y2
```

Reply:
Oh, yes :dizzy: 
(By the way, this code isn't called on the last commit, but is good to prepare it anyway)

Reply:
follow up question, is there a missing unittest that would have caught that mistake?

Reply:
There is a test but the cases covered aren't enough